pkg/master/master.go

```diff
@@ -778,18 +796,25 @@ func (m *Master) thirdpartyapi(group, kind, version, pluralResource string) *api
 	}
 }
 
-func (m *Master) GetRESTOptionsOrDie(c *Config, resource unversioned.GroupResource) generic.RESTOptions {
-	storageConfig, err := c.StorageFactory.NewConfig(resource)
+type restOptionsFactory struct {
+	deleteCollectionWorkers int
+	enableGarbageCollection bool
+	storageFactory          genericapiserver.StorageFactory
+	storageDecorator        generic.StorageDecorator
+}
+
+func (f restOptionsFactory) GetRESTOptionsOrDie(resource unversioned.GroupResource) generic.RESTOptions {
```

Comment:
Why is this not called `NewFor` ?

Any reason against sharing the RestOpenFactory between federation and apiserver? Not in genericapiserver of course.